sql/core/src/test/scala/org/apache/spark/sql/test/DataFrameReaderWriterSuite.scala

```diff
@@ -225,21 +225,13 @@ class DataFrameReaderWriterSuite extends QueryTest with SharedSQLContext with Be
   }
 
   test("pass partitionBy as options") {
-    Seq(true, false).foreach { flag =>
-      withSQLConf(SQLConf.LEGACY_PASS_PARTITION_BY_AS_OPTIONS.key -> s"$flag") {
-        Seq(1).toDF.write
-          .format("org.apache.spark.sql.test")
-          .partitionBy("col1", "col2")
-          .save()
-
-        if (flag) {
-          val partColumns = LastOptions.parameters(DataSourceUtils.PARTITIONING_COLUMNS_KEY)
-          assert(DataSourceUtils.decodePartitioningColumns(partColumns) === Seq("col1", "col2"))
-        } else {
-          assert(!LastOptions.parameters.contains(DataSourceUtils.PARTITIONING_COLUMNS_KEY))
-        }
-      }
-    }
+    Seq(1).toDF.write
+      .format("org.apache.spark.sql.test")
+      .partitionBy("col1", "col2")
+      .save()
+
+    val partColumns = LastOptions.parameters(DataSourceUtils.PARTITIONING_COLUMNS_KEY)
+    assert(DataSourceUtils.decodePartitioningColumns(partColumns) === Seq("col1", "col2"))
```

Comment:
`decodePartitioningColumns` is under `execution` package that's not supposed to be exposed so users shouldn't use this util directly.

Did we document this option to any public datasource v1 API? We should also say this is a JSON string.